contrib/hashicorp/vault/option.go

```diff
@@ -0,0 +1,38 @@
+package vault
+
+import "math"
+
+type config struct {
+	withAnalytics bool
+	analyticsRate float64
+	serviceName   string
+}
+
+// Option can be passed to NewHTTPClient and WrapHTTPClient to configure the integration.
+type Option func(*config)
+
+func defaults(cfg *config) {
+	cfg.serviceName = "vault"
+	cfg.analyticsRate = math.NaN()
+}
+
+// WithAnalytics enables or disables Trace Analytics for all started spans.
+func WithAnalytics(on bool) Option {
+	return func(c *config) {
```

Comment:
```go
if on {
    return WithAnalyticsRate(1.0)
} else {
    return WithAnalyticsRate(math.NaN())
}
```
Let's make sure to test all scenarios:
* Analytics on, rate tag is set at 1.0
* Analytics off, no rate tag is set
* Analytic rate set, rate tag set

etc.

Reply:
Done.